staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt.go

```diff
@@ -159,84 +396,192 @@ func (tr *TypedConfigObjectReference) String() string {
+func rmSetup(loopbackClientConfig *restclient.Config, serverConcurrencyLimit int, requestWaitLimit time.Duration) *requestManagement {
+	reqMgmt := &requestManagement{
 		configQueue:            workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(200*time.Millisecond, 8*time.Hour), "req_mgmt_config_queue"),
 		serverConcurrencyLimit: serverConcurrencyLimit,
 		requestWaitLimit:       requestWaitLimit,
 	}
 	// TODO: finish implementation
+
+	// TODO(aaron-prindle) figure out what needs to be done...
+	// fetch related k8s objects - FlowSchemas, RequestPriorityLevels
+	return reqMgmt
+}
+
+func initPriorityLevelStates(plcs []*rmtypesv1a1.PriorityLevelConfiguration, requestWaitLimit time.Duration) map[string]*PriorityLevelState {
+	pls := map[string]*PriorityLevelState{}
+
+	// TODO(aaron-prindle) change this to a test method and change clock to be
+	// clk := clock.RealClock{}
+	clk := clock.RealClock{}
+
+	for _, plc := range plcs {
+
+		pls[plc.Name] = &PriorityLevelState{
+			config: plc.Spec,
+			fqs: NewFairQueuingSystem(int(plc.Spec.Queues),
+				int(plc.Spec.QueueLengthLimit), requestWaitLimit, clk),
+		}
+	}
+	return pls
 }
 
 // WithRequestManagement limits the number of in-flight requests in a fine-grained way
 func WithRequestManagement(
 	handler http.Handler,
-	loopbackClientConfig *restclient.Config,
+	// loopbackClientConfig *restclient.Config,
+	// -- added, REMOVE
+	flowSchemas []*rmtypesv1a1.FlowSchema,
+	priorityLevelConfigurations []*rmtypesv1a1.PriorityLevelConfiguration,
+	// --
 	serverConcurrencyLimit int,
 	requestWaitLimit time.Duration,
 	longRunningRequestCheck apirequest.LongRunningRequestCheck,
 ) http.Handler {
-	rmStartOnce.Do(func() { rmSetup(loopbackClientConfig, serverConcurrencyLimit, requestWaitLimit) })
-
+	// reqMgmt := rmSetup(loopbackClientConfig, serverConcurrencyLimit, requestWaitLimit)
+	// --
+	pls := initPriorityLevelStates(priorityLevelConfigurations, requestWaitLimit)
+	rmState := RMState{
+		flowSchemas:            flowSchemas,
+		priorityLevelStates:    pls,
+		serverConcurrencyLimit: serverConcurrencyLimit,
+	}
+	// --
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		ctx := r.Context()
-		requestInfo, ok := apirequest.RequestInfoFrom(ctx)
-		if !ok {
-			handleError(w, r, fmt.Errorf("no RequestInfo found in context, handler chain must be wrong"))
-			return
-		}
+		// =======================
+		// TODO(aaron-prindle) CHANGE removed for early testing
+		// ctx := r.Context()
+		// requestInfo, ok := apirequest.RequestInfoFrom(ctx)
+		// if !ok {
+		// 	handleError(w, r, fmt.Errorf("no RequestInfo found in context, handler chain must be wrong"))
+		// 	return
+		// }
 
 		// Skip tracking long running events.
-		if longRunningRequestCheck != nil && longRunningRequestCheck(r, requestInfo) {
-			handler.ServeHTTP(w, r)
-			return
-		}
+		// if longRunningRequestCheck != nil && longRunningRequestCheck(r, requestInfo) {
```

Comment:
Why comment this out?  It is easy enough to pass `nil` in a test.

Reply:
updated